src/power_grid_model/validation/utils.py

```diff
@@ -107,7 +101,7 @@ def _update_input_data(input_data: SingleDataset, update_data: SingleDataset):
 
 
 def _update_component_data(
-    component: _ComponentTypeLike, input_data: SingleComponentData, update_data: SingleComponentData
+    component: ComponentType, input_data: SingleComponentData, update_data: SingleComponentData
```

Comment:
here and elsewhere: please double-check that this is not backwards incompatible for users

Reply:
Yes! All public functions should have a conversion to enum . This particular one is covered at validate_input_data and validate_update_data